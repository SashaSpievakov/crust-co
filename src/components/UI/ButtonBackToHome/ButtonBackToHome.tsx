import { Link } from 'react-router-dom';

import { ButtonBack } from '../../../styles/Buttons.styled';

const ButtonBackToHome = () => {
  return (
    <ButtonBack to="/" as={Link}>
      <span>Go Back</span>
    </ButtonBack>
  );
};

export default ButtonBackToHome;
