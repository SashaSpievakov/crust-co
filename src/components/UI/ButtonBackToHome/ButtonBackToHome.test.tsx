import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import ButtonBackToHome from './ButtonBackToHome';
import renderWithProviders from '../../../tests/helpers/renderWithProviders';
import rendererWithProviders from '../../../tests/helpers/rendererWithProviders';

describe('ButtonBackToHome Tests', () => {
  test('renders the ButtonBackToHome UI component', () => {
    const snapshot = rendererWithProviders(<ButtonBackToHome />);
    expect(snapshot).toMatchSnapshot();
  });

  test('checks redirect to the Home page', () => {
    renderWithProviders(null, true, '/testing');
    const link = screen.getByRole('link', {
      name: /go back/i,
    });

    userEvent.click(link);
    expect(screen.getByText(/all pizzas/i)).toBeInTheDocument();
  });
});
