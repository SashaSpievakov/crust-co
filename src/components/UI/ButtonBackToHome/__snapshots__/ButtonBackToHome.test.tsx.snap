// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBackToHome Tests renders the ButtonBackToHome UI component 1`] = `
Object {
  "children": Array [
    <span>
      Go Back
    </span>,
  ],
  "props": Object {
    "className": "sc-bcXHqe sc-gswNZR fAfvZN llPphw",
    "href": "/",
    "onClick": [Function],
    "target": undefined,
  },
  "type": "a",
}
`;
